src/sitemap/handler.js

```diff
@@ -393,8 +394,78 @@ export async function sitemapAuditRunner(baseURL, context) {
   };
 }
 
+async function convertToOpportunity(auditUrl, auditData, context) {
+  const { dataAccess, log } = context;
+
+  const opportunities = await dataAccess.Opportunity.allBySiteIdAndStatus(auditData.siteId, 'NEW');
+  let sitemapOpptyPagesWithIssues = opportunities.find((oppty) => oppty.getType() === 'sitemap-pages-with-issues');
+
+  const sitemapsWithPagesWithIssues = auditData.auditResult?.details?.issues
+    ? Object.keys(auditData.auditResult?.details?.issues)
+    : [];
+
+  const newData = [];
+  sitemapsWithPagesWithIssues.forEach((sitemapUrl) => {
+    const pagesWithIssues = auditData.auditResult.details.issues[sitemapUrl];
+    pagesWithIssues.forEach((page) => {
+      newData.push({
+        ...page,
+        sitemapUrl,
+      });
+    });
+  });
+
+  if (newData.length > 0) {
+    if (!sitemapOpptyPagesWithIssues) {
+      const opportunityData = {
+        siteId: auditData.siteId,
+        auditId: auditData.id,
+        runbook: 'link-to-runbook',
+        type: 'sitemap-pages-with-issues',
+        origin: 'AUTOMATON',
+        title: 'Opportunity Title',
+        description: 'Opportunity Description',
+        guidance: {
+          steps: [
+            'Step 1',
+            'Step 2',
+          ],
+        },
+        tags: ['indexability', 'sitemap'],
+      };
+      sitemapOpptyPagesWithIssues = await dataAccess.Opportunity.create(opportunityData);
+    } else {
+      sitemapOpptyPagesWithIssues = sitemapOpptyPagesWithIssues.setAuditId(auditData.id);
+    }
+  }
+
+  if (newData.length > 0 && sitemapOpptyPagesWithIssues) {
+    const buildKey = (auditData1) => `${auditData1.siteId}|sitemap-pages-with-issues|${auditData1.auditResult.reasons.map((a) => a.value)};}`;
```

Comment:
Here, we need to distinguish each suggestion entry (i.e. each issue in the Sitemap)

So, while building the key, we should also use the "sitemap url" and "page url" from the audit result, when they exist (i.e. success=true)